repos/system_upgrade/common/actors/detectgrubconfigerror/tests/test_detectgrubconfigerror.py

```diff
@@ -0,0 +1,58 @@
+from leapp.models import GrubConfigError, Report
+from leapp.utils import report
+
+grub_cmdline_syntax_error = GrubConfigError(error_type=GrubConfigError.ERROR_GRUB_CMDLINE_LINUX_SYNTAX,
+                                            files=['/etc/default/grub.cfg'])
+grub_cmdline_syntax_error2 = GrubConfigError(error_type=GrubConfigError.ERROR_GRUB_CMDLINE_LINUX_SYNTAX,
+                                             files=['/boot/grub2/grub.cfg', '/etc/default/someothergrub.cfg'])
+
+grub_missing_newline_error = GrubConfigError(error_type=GrubConfigError.ERROR_MISSING_NEWLINE,
+                                             files=['/etc/default/someothergrub.cfg'])
+grub_missing_newline_error2 = GrubConfigError(error_type=GrubConfigError.ERROR_MISSING_NEWLINE,
+                                              files=['/etc/default/grub'])
+
+grub_corrupted_config = GrubConfigError(error_type=GrubConfigError.ERROR_CORRUPTED_GRUBENV,
+                                        files=['/boot/grub2/grub.cfg', '/boot/efi/EFI/redhat/grub.cfg'])
+grub_corrupted_config2 = GrubConfigError(error_type=GrubConfigError.ERROR_CORRUPTED_GRUBENV,
+                                         files=['/boot/grub2/grub.cfg'])
+
+
+def test_cmdline_syntax_error(current_actor_context):
+    # Make sure that just 1 low priority report message is created with config files present.
+    current_actor_context.feed(grub_cmdline_syntax_error)
+    current_actor_context.feed(grub_cmdline_syntax_error2)
+    current_actor_context.run()
+    messages = current_actor_context.consume(Report)
+    assert len(messages) == 1
+    message = messages[0]
+    assert 'Syntax error detected in grub configuration' in message.report['title']
+    assert message.report['severity'] == 'low'
+    assert message.report['detail']['related_resources'][0]['title'] == '/etc/default/grub.cfg'
+
+
+def test_missing_newline(current_actor_context):
+    # Make sure that just 1 low priority report message is created with config files present
+    current_actor_context.feed(grub_missing_newline_error)
+    current_actor_context.feed(grub_missing_newline_error2)
+    current_actor_context.run()
+    messages = current_actor_context.consume(Report)
+    assert len(messages) == 1
+    message = messages[0]
+    assert 'Detected a missing newline at the end of grub configuration file' in message.report['title']
+    assert message.report['severity'] == 'low'
+    assert message.report['detail']['related_resources'][0]['title'] == '/etc/default/someothergrub.cfg'
+
+
+def test_corrupted_config(current_actor_context):
+    # Make sure that just 1 low priority report message is created with config files present
```

Comment:
```suggestion
    # Make sure that just 1 high priority report message is created with config files present
```

Reply:
If I understand correctly, it is not expected that actor will receive multiple messages of the same error type. There might be a problem in the future with these tests that they expect a certain ordering on how the actor processes the messages. If the actor implementation slightly changes for whatever reason and instead of the first message of a given type, the last one will be used to generate the report, the person implementing the changes will have to change it.

That being said, these are just tests and the above comment is meant to maybe give you a perspective you might consider in the future. Maybe I would feed the actor only with one message, but otherwise the tests are fine. We have more important things to work on then to make absolutely flawless unit tests.

Reply:
The only expectation so far is that the actor produces not more than 1 message of a certain type, yeah. 
I added this "1 message only check" here specifically to check that our current convention "only 1 message" is being met. If for some reason it changes in the future then these tests will fail, right, but that is the idea - to make the convention visible.
If it's ok with you I'd rather keep things as they are now, I really like visibility.